### DISCUSSION TITLE
Unused code setting channel ignore mode

### DISCUSSION THREAD

**Discussion by d3x0r** [libssh2/libssh2#1344] [Ideas]
Was going through the code and found some unused code.

https://github.com/libssh2/libssh2/blob/master/src/channel.c#L2009-L2017

if `channel->extData2_state == libssh2_NB_state_idle` when the function starts, then the first if changes it to `libssh2_NB_state_created` and then for the next if the state won't be idle anymore, and the second if won't run.  If it's not idle, then nothing happens, and the state is reset to idle.

Really this is the only function that uses `extData2_state`

So the function becomes just

```
        _libssh2_debug((channel->session, LIBSSH2_TRACE_CONN,
                       "Setting channel %u/%u handle_extended_data"
                       " mode to %d",
                       channel->local.id, channel->remote.id, ignore_mode));
        channel->remote.extended_data_ignore_mode = (char)ignore_mode;
```

and `extData2_state` can be removed from the channel structure.

---

Or - maybe if there is really a flush still desired, maybe the second if should use `channel->extData2_state == libssh2_NB_state_created` instead of `channel->extData2_state == libssh2_NB_state_idle`.

**Comment by vszakats** [comment]
@d3x0r Thanks for your report. Would you mind making a PR out of this?

**Comment by d3x0r** [comment]
to include or exclude the flush?

**Comment by vszakats** [comment]
The title says there is unused code, so I suppose to exclude.
If there is a different issue, to address that.

A PR would allow the tests to run and team members to review.

If there no definite action at this time, we can move this to
discussions.

**Comment by d3x0r** [comment]
Hmm I don't see a back reference automatic... so PR #1345
